### PR TITLE
[REEF-718] Document EvaluatorBatchId

### PR DESCRIPTION
**Pull request by afchung**
* Documents EvaluatorBatchId and field related to it.
	* Cleaned up some code relevant to EvaluatorBatchId.

JIRA:
  [REEF-718] https://issues.apache.org/jira/browse/REEF-718

Pull Request:
  Closes #

### CODE REVIEW

File: lang/cs/Org.Apache.REEF.Driver/Bridge/Events/AllocatedEvaluator.cs [hunk 1]
```diff
@@ -173,18 +173,12 @@ private void ProcessNewEvaluator()
...
-                            "Received evalautor [{0}] of memory {1}MB that matches request of {2}MB with batch id [{3}].",
-                            Id,
-                            _evaluatorDescriptor.Memory,
-                            pair.Value.Memory,
-                            assignedId);
-
-                        LOGGER.Log(Level.Verbose, message);
+                        var assignedId = key.Substring(0, key.LastIndexOf(EvaluatorRequestor.BatchIdxSeparator));
+
+                        LOGGER.Log(Level.Verbose, "Received evalautor [{0}] of memory {1}MB that matches request of {2}MB with batch id [{3}].",
```
Comment (jwang98052) [inline lang/cs/Org.Apache.REEF.Driver/Bridge/Events/AllocatedEvaluator.cs]: typo evalautor

File: lang/cs/Org.Apache.REEF.Driver/Bridge/Events/EvaluatorRequestor.cs [hunk 2]
```diff
@@ -33,9 +34,10 @@ namespace Org.Apache.REEF.Driver.Bridge.Events
     [DataContract]
     internal class EvaluatorRequestor : IEvaluatorRequestor
     {
+        internal const char BatchIdxSeparator = '_';
         private static readonly Logger LOGGER = Logger.GetLogger(typeof(EvaluatorRequestor));
-
-        private static Dictionary<string, IEvaluatorDescriptor> _evaluators;
+        private static readonly Lazy<IDictionary<string, IEvaluatorDescriptor>> EvaluatorsInstantiator =
```
Comment (jwang98052) [inline lang/cs/Org.Apache.REEF.Driver/Bridge/Events/EvaluatorRequestor.cs]: Would it be much simpler to create the Dictionary at this line itself? 
private static readonly IDictionary<string, IEvaluatorDescriptor> EvaluatorsInstantiator = new Dictionary<string, IEvaluatorDescriptor>()
We need this dictionary anyway and creating an empty dictionary at static time has no any risk.

File: lang/cs/Org.Apache.REEF.Driver/Evaluator/EvaluatorRequestBuilder.cs [hunk 3]
```diff
@@ -93,7 +93,11 @@ public EvaluatorRequestBuilder SetRackName(string rackName)
             return this;
         }
 
-        // TODO[REEF-718]: Document.
+        /// <summary>
```
Comment (jwang98052) [inline lang/cs/Org.Apache.REEF.Driver/Evaluator/EvaluatorRequestBuilder.cs]: Here_evaluatorBatchId is per builder therefore per request. How do you ensure all requests in the same batch share the same bath id?

**Comment by jwang98052** [comment]
I completed my pass.

**Comment by afchung** [comment]
@jwang98052 Addressed comments. Please have another pass. Thanks!

**Comment by jwang98052** [comment]
I will test and merge

**Pull request closed without merging**